src/CodeGen/ODataT4CodeGenerator.ttinclude

```diff
@@ -317,6 +327,70 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string stringValue) 

```

Comment:
Ideally, functions should only do one thing, just the name ValidateAndSetCustomHttpHeadersFromString(string stringValue) suggests the function may be doing too much.

Reply:
I have split the functions